examples/todomvc/app.re

```diff
@@ -36,7 +38,10 @@ module Top = {
       None
     };
     let handleChange {state} event =>
-      Some {...state, newTodo: ReasonJs.Document.value event##target};
+      switch (event##target |> Element.asHtmlElement) {
```

Comment:
I don't think using functors would mean consumers would have to relate to them. I don't even know if this would work, but the idea is that you'd using functors as mixins to include implementations in a module. E.g., like in this case, of document and node was implemented as functors (but exposed as ordinary modules), then HtmlDocument could perhaps include all the functions of Document, and Element include all the functions of Node by including the respective functors.

But hhis is of course based on my very limited understaning of functors, including and the type system generally.